### PR TITLE
fix #303611: Adding fixed spacer can clash with vertical frame below, causing a crash.

### PR DESCRIPTION
**Pull request by Jojo-Schmitz**
Solves https://musescore.org/en/node/303611 and hopefully doesn't break other cases

### CODE REVIEW

File: libmscore/system.cpp [hunk 1]
```diff
@@ -1103,7 +1103,7 @@ void System::scanElements(void* data, void (*func)(void*, Element*), bool all)
 qreal System::staffYpage(int staffIdx) const
       {
       if (_staves.size() <= staffIdx || staffIdx < 0) {
-            qFatal("staffY: staves %d: bad staffIdx %d", _staves.size(), staffIdx);
+            qWarning("staffY: staves %d: bad staffIdx %d", _staves.size(), staffIdx);
```
Comment (igorkorsukov) [inline libmscore/system.cpp:1106]: I think, need to find and fix the reason why staffIdx is not valid  
And in this place add assert

Comment (Jojo-Schmitz) [reply libmscore/system.cpp]: like the next commit?

Comment (igorkorsukov) [reply libmscore/system.cpp]: Yes :)   
But there is a problem, why is an invalid index passed?

Comment (Jojo-Schmitz) [reply libmscore/system.cpp]: I guess because there is no staff, just a frame

Comment (igorkorsukov) [reply libmscore/system.cpp]: Then I think this method should not be called, in such cases.

**Comment by Jojo-Schmitz** [comment]
Hmm, why do the vtests fail now?

**Comment by AntonioBL** [comment]
> 
> 
> Hmm, why do the vtests fail now?

I don't know. It seems that the png images generated by one of the branches being compared are corrupted.
Could you please rebase?

**Comment by Jojo-Schmitz** [comment]
$ git pull --rebase musescore master
    From https://github.com/musescore/MuseScore
     * branch                master     -> FETCH_HEAD
    Current branch spacer-crash is up to date.

**Comment by AntonioBL** [comment]
Ok. I will try to reproduce under Linux.
And in the meantime I have a PR  ready for submission based on the current master; we'll see how it behaves.

**Comment by Jojo-Schmitz** [comment]
Mine isn't good enough anyhow

**Comment by AntonioBL** [comment]
The vtests in my PR worked, so it does not seem a problem of that master branch commit.
Maybe it was just some random quirk of Github servers.

**Comment by AntonioBL** [comment]
Also normal mtests fail on Travis throwing an exception:
https://travis-ci.org/github/musescore/MuseScore/jobs/672969457#L4314

**Comment by AntonioBL** [comment]
Shouldn't the condition inside the IF_ASSERT_FAILED be negated?

**Comment by Jojo-Schmitz** [comment]
Well possible. It just crashes this way, most probably the cause for the failing v- and mtests

**Comment by AntonioBL** [comment]
Is maybe a "return" missing?

**Comment by Jojo-Schmitz** [comment]
Darn....

**Comment by Jojo-Schmitz** [comment]
That doesn't fix the crash of the issue at hand though

**Comment by Jojo-Schmitz** [comment]
Hmm, well... the 1st commit fixes the crash, the 2nd makes it fail at the same place (except in a Release build, I think)

**Pull request merged**